File: quantly-trading-service/models/account.go
package models

import (
	"encoding/json"
	"time"
)

type Account struct {
	AccountId              string          `json:"account_id" gorm:"primaryKey;type:varchar(64)"`
	UserId                 string          `binding:"required" json:"user_id" gorm:"type:varchar(64)"`
	AccountNumber          string          `json:"account_number"`
	Status                 string          `json:"status"`
	CryptoStatus           string          `json:"crypto_status"`
	Currency               string          `json:"currency"`
	LastEquity             string          `json:"last_equity"`
	CreatedAt              time.Time       `json:"created_at"`
	AccountType            string          `json:"account_type"`
	EnabledAssets          json.RawMessage `json:"enabled_assets"` // JSON array of enabled assets
	PartnerUserId          string          `json:"partner_user_id"`
	FundingInstructionsUrl string          `json:"funding_instructions_url"`
	PatternDayTrader       int8            `json:"pattern_day_trader"` // 0 or 1
	KycCompletedAt         time.Time       `json:"kyc_completed_at"`
	KycStatus              string          `json:"kyc_status"`
	AccountAtype           string          `json:"account_atype"`
	ManagementStatus       string          `json:"management_status"`
	ClearingBroker         string          `json:"clearing_broker"`
	ClearingAccountNumber  string          `json:"clearing_account_number"`
	Contacts               []Contact       `gorm:"foreignKey:AccountId;references:AccountId"`
}

File: quantly-stock-data-service/models/account.py
import uuid

from sqlmodel import SQLModel, Field
from typing import Optional
from datetime import datetime

class Account(SQLModel, table=True):
    __tablename__ = "accounts"

    account_id: str = Field(primary_key=True, max_length=36)
    account_number: Optional[str] = Field(default=None, max_length=36)
    status: Optional[str] = Field(default=None, max_length=10)
    currency: Optional[str] = Field(default=None, max_length=10)
    last_equity: Optional[str] = Field(default=None, max_length=20)
    create_at: Optional[datetime] = None
    account_type: Optional[str] = Field(default=None, max_length=20)
    trading_type: Optional[str] = Field(default=None, max_length=20)
    user_id: Optional[str] = Field(default=None, foreign_key="users.user_id", max_length=36)
